litellm/llms/ollama.py

```diff
@@ -212,24 +212,29 @@ def get_ollama_response(
 
     ## RESPONSE OBJECT
     model_response["choices"][0]["finish_reason"] = "stop"
-    if optional_params.get("format", "") == "json":
+    if data.get("format", "") == "json":
```

Comment:
@ChristianWeyer since we now allow format:json as a param (mapping the openai 'response_format' param),

i think this condition (even our existing one) is probably invalid. 

We might need to do this like anthropic, where we propogate the tool calling params till ollama.py, and set a bool value (e.g. `_is_function_call`) to check if the user is explicitly trying to make a function call, before setting params. 

Let me know if you can help with this

Reply:
How would we determine whether the user is making a Function call?